app/src/forms/common/scheduleService.js

```diff
@@ -0,0 +1,597 @@
+  let timezone = 'America/Vancouver';
+
+  if (datesOrSchedule && datesOrSchedule.scheduleType !== undefined) {
+    const schedule = datesOrSchedule;
+    dates = getSubmissionPeriodDates(schedule);
+    allowLateSubmissions = schedule.allowLateSubmissions && schedule.allowLateSubmissions.enabled;
+    timezone = schedule.timezone || 'America/Vancouver';
+  } else {
+    dates = datesOrSchedule;
+    // Try to extract timezone if available
+    if (dates && dates.length > 0 && dates[0].timezone) {
+      timezone = dates[0].timezone;
+    }
+  }
+
+  // No dates, return null
+  if (!dates || dates.length === 0) return null;
+
+  const { openDate, closeDate, graceDate } =
+    dates[0].scheduleType !== undefined
+      ? extractScheduleDates(dates[0])
+      : {
+          openDate: dates[0].startDate,
+          closeDate: dates[0].closeDate,
+          graceDate: dates[0].graceDate,
+        };
+
+  // Manual schedule type (no closing date)
+  if (!closeDate) {
+    return {
+      state: 1,
+      index: 0,
+      dates: dates[0],
+      old_dates: null,
+      late: 0,
+    };
+  }
+
+  // Today's date
+  const today = referenceDate || new Date();
+  const ignoreTime = !respectTimeComponent;
+
+  // End date is grace date if late submissions allowed, otherwise close date
+  const endDate = allowLateSubmissions && graceDate ? graceDate : closeDate;
+
+  // Check if today is within period
+  if (isDateInRange(today, openDate, endDate, timezone, ignoreTime)) {
+    // Check if in late submission period
+    const isLate = allowLateSubmissions && closeDate && !isDateInRange(today, openDate, closeDate, timezone, ignoreTime);
+
+    return {
+      state: 1,
+      index: 0,
+      dates: dates[0],
+      old_dates: null,
+      late: isLate ? 1 : 0,
+    };
+  }
+
+  // Not in period, determine if before or after
+  const isBeforePeriod = isDateInFuture(openDate, timezone, today, ignoreTime);
+
+  return {
+    state: isBeforePeriod ? -1 : 0,
+    index: -1,
+    dates: dates[0], // Return the period as a reference
+    old_dates: null,
+    late: -1,
+  };
+}
+
+/**
+ * Determines what type of reminder email should be sent based on current date and schedule
+ * @param {Object} scheduleOrReport The form schedule or period report
+ * @param {Date|String} referenceDate Optional reference date (defaults to now)
+ * @param {Boolean} respectTimeComponent Whether to respect time components in dates (defaults to false)
+ * @returns {String|undefined} Email type constant or undefined if no reminder should be sent
+ */
+function getEmailReminderType(scheduleOrReport, referenceDate = null, respectTimeComponent = false) {
+  // Get the current period if a schedule was provided
+  const report = scheduleOrReport.scheduleType !== undefined ? getCurrentPeriod(scheduleOrReport, referenceDate, respectTimeComponent) : scheduleOrReport;
+
+  if (!report || !report.dates) return undefined;
+
+  const now = referenceDate || new Date();
```

Comment:
make sure you are doing current or now the same everywhere. i see multiple different ways of setting now/current. should probably make a function you can export from scheduleService and use in reminderService.